src/cli/migrate/nodeVersion/getNodeTypesVersion.ts

```diff
@@ -0,0 +1,56 @@
+import npmFetch from 'npm-registry-fetch';
+import { gt, satisfies, valid } from 'semver';
+import { z } from 'zod';
+
+type VersionResult = {
+  version: string;
+  err?: string;
+};
+
+const NpmFetchResponse = z.object({
+  versions: z.record(
+    z.string(),
+    z.object({
+      name: z.string(),
+      version: z.string(),
+      deprecated: z.string().optional(),
+    }),
+  ),
+});
+
+export const getNodeTypesVersion = async (
+  major: number,
+  defaultVersion: string,
+): Promise<VersionResult> => {
+  try {
+    const response = await npmFetch.json('@types/node', {
+      headers: {
+        Accept:
+          'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*',
+      },
+    });
+
+    const parsedVersions = NpmFetchResponse.safeParse(response);
```

Comment:
Out of curiosity, is this noticeably slow in your testing? The response seems massive 😂